packages/core/src/utils.ts

```diff
@@ -56,7 +56,9 @@ export function toStatePath(stateId: string | string[]): string[] {
     return stateId;
   }
 
-  return stateId.split(STATE_DELIMITER);
+  // Split on periods, but not if they are escaped
+  // i.e. match a period only if it is not preceded by a backslash.
+  return stateId.split(/(?<!\\)\./g);
```

Comment:
note: a negative lookbehind requires ES2018

Reply:
Correct me if I'm wrong but this doesn't look correct.

An escaping character escapes the *next* character so the next character loses its special meaning. So while `\.` escapes a dot, this `\\.` should escape the backslash character and the dot should stay unescaped.

Since we are going between languages~ (JS -> XState), in the source code that would actually be `\\\\.` (just how `\.` in the source code is `\\.`)